python/ray/serve/tests/test_deploy.py

```diff
@@ -424,13 +424,15 @@ def make_nonblocking_calls(expected, expect_blocking=False):
         # Returns dict[val, set(pid)].
         blocking = []
         responses = defaultdict(set)
+        timeout_value = 500 if sys.platform == "win32" else 30
         start = time.time()
-        while time.time() - start < 30:
+        while time.time() - start < timeout_value:
             refs = [call.remote(block=False) for _ in range(10)]
             ready, not_ready = ray.wait(refs, timeout=5)
             for ref in ready:
                 val, pid = ray.get(ref)
-                responses[val].add(pid)
+                if val != "<NULL>" and pid != "<NULL>":
+                    responses[val].add(pid)
```

Comment:
I have added this check so that behaviour before [this change](https://github.com/ray-project/ray/pull/21391/files#diff-b2841aa657072ac1c56e96e3ccc7edd327920ad41a10606bf1403b84ae5c1c51R398) is retained without throwing into `IndexError`.